sdk/compiler/damlc/lib/DA/Cli/Damlc/Command/MultiIde.hs

```diff
@@ -357,65 +357,55 @@ sendAllSubIDEs miState msg = holdingIDEsAtomic miState $ \ides ->
 sendAllSubIDEs_ :: MultiIdeState -> LSP.FromClientMessage -> IO ()
 sendAllSubIDEs_ miState = void . sendAllSubIDEs miState
 
-getSourceFileHome :: MultiIdeState -> FilePath -> STM (Maybe PackageHome)
+getDirectoryIfFile :: FilePath -> IO FilePath
+getDirectoryIfFile path = do
+  isFile <- doesFileExist path
+  pure $ if isFile then takeDirectory path else path
+
+getSourceFileHome :: MultiIdeState -> FilePath -> STM PackageHome
 getSourceFileHome miState path = do
+  -- If the path is a file, we only care about the directory, as all files in the same directory share the same home
+  dirPath <- unsafeIOToSTM $ getDirectoryIfFile path
   sourceFileHomes <- takeTMVar (sourceFileHomesVar miState)
```

Comment:
Changed this cache to hold the directory name, rather than file name, to avoid many IO lookups when moving an IDE